internal/server/boltdbstate/task.go

```diff
@@ -79,8 +79,51 @@ func (s *State) TaskDelete(ref *pb.Ref_Task) error {
 	return err
 }
 
+// GetJobsByTaskRef will look up every job triple by Task ref in a single
+// memdb transaction. This is often used via the API for building out
+// a complete picture of a task beyond the job ID refs.
+func (s *State) GetJobsByTaskRef(
+	task *pb.Task,
+) (startJob *pb.Job, taskJob *pb.Job, stopJob *pb.Job, err error) {
+	memTxn := s.inmem.Txn(true)
+	defer memTxn.Abort()
+
+	err = s.db.View(func(dbTxn *bolt.Tx) error {
+		resp, err := s.JobById(task.StartJob.Id, nil)
```

Comment:
Nesting read locks is _fine_ but weird. This is why we usually have the unexported versions that take the transaction. If `jobById` doesn't exist, I'd introduce that, and pass in `dbTxn` instead.

Reply:
That's true. Yeah it makes a lot more sense! Good call. `jobById` does exist so I'll use that instead

Reply:
Addressed in ab18de5de2757fd3a519b5277b080a6201376a0a